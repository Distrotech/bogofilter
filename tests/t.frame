# testing framework for /bin/sh

set -e
TMPDIR=`mktemp -d ./checks.$$.XXXXXX`
trap "rm -r ${TMPDIR}" 0
BOGOFILTER_DIR=$TMPDIR
export BOGOFILTER_DIR
../bogofilter -n </dev/null
