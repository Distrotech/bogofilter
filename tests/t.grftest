#!/bin/sh

. ${srcdir:=.}/t.frame

set -e

# Note:  This test DOES leave its output files on the hard drive.
#	 This makes it easier to see what has happened, which is
#		especially useful when something is different.
#
#        In directory t.systest.d are:
#
#		tests   - directory containing individual output files
#		grftest.out - comparison of directories:
#			      outputs - known correct outputs
#			      tests   - outputs of the individual tests

verbose=0
if [ "$1" = "-v" ]; then
    verbose=1
fi

[   -d ${TMPDIR} ] && rm -f ${TMPDIR}/*
[ ! -d ${TMPDIR} ] && mkdir ${TMPDIR}

cat <<EOF > ${TMPDIR}/cfg.r
robx=0.415
min_dev=0.0
EOF

cat <<EOF > ${TMPDIR}/cfg.f
robx=0.415
min_dev=0.1
EOF

cat /dev/null > ${TMPDIR}/cfg.g

BOGOUTIL="../bogoutil"
BOGOFILTER="../bogofilter"
SYSTEST="${srcdir}/t.systest.d"

map_rc()
{
    (
	set +e
	eval "$@"
	a=$?
	[ $a -eq 0 ] && exit 0
	[ $a -eq 1 ] && exit 0
	exit $a
    )
}

run_test()
{
    id="$1"
    out="$2"
    ver="$3"
    OUT="${TMPDIR}/tests/msg.$out.$alg.$ver"
    OPTS="-$alg -t -$ver $CFG"
    map_rc $BOGOFILTER $OPTS <${SYSTEST}/inputs/msg.$id.txt >${TMPDIR}/tests-$alg 
    sed s/,.version=.*// < ${TMPDIR}/tests-$alg > $OUT
}

[   -d ${TMPDIR}/tests ] && rm -f ${TMPDIR}/tests/*
[ ! -d ${TMPDIR}/tests ] && mkdir -p ${TMPDIR}/tests

for alg in r f g ; do
    BOGOFILTER_DIR="${TMPDIR}/tests.$alg"
    CFG="-c ${TMPDIR}/cfg.$alg"
    export BOGOFILTER_DIR
    [   -d $BOGOFILTER_DIR ] && rm -f $BOGOFILTER_DIR/*
    [ ! -d $BOGOFILTER_DIR ] && mkdir $BOGOFILTER_DIR
    $BOGOFILTER -$alg $CFG -s < ${SYSTEST}/inputs/spam.mbx
    $BOGOFILTER -$alg $CFG -n < ${SYSTEST}/inputs/good.mbx
    #
    # run tests for msg.[1-8].txt
    #
    for msg in ${SYSTEST}/inputs/msg.?.txt ; do
	tst=`echo $msg | sed s@${SYSTEST}/inputs/msg.@@ | sed s@.txt@@`
	args="$tst $tst v"
	run_test $args
    done

    rm -f ${TMPDIR}/tests-$alg
done

OUT="grftest.out"

printf "%2s     %-10s   %-10s   %-10s\n" id graham robinson fisher > ${TMPDIR}/tests/$OUT
for out in ${TMPDIR}/tests/msg.?.r.v ; do
    id=`echo $out | sed s@${TMPDIR}/tests/msg.@@ | sed s@.r.v@@`
    g=`cat ${TMPDIR}/tests/msg.$id.g.v | tr "NY" "hS"`
    r=`cat ${TMPDIR}/tests/msg.$id.r.v | tr "NY" "hS"`
    f=`cat ${TMPDIR}/tests/msg.$id.f.v | tr "NY" "hS"`
    printf "%2s   %c %s   %c %s   %c %s\n" $id $g $r $f >> ${TMPDIR}/tests/$OUT
done

if  [ $verbose -eq 1 ]; then
    diff -u -s ${SYSTEST}/outputs/$OUT ${TMPDIR}/tests/$OUT
fi

cmp ${SYSTEST}/outputs/$OUT ${TMPDIR}/tests/$OUT
