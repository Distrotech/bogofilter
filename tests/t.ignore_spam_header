#! /bin/sh

. ${srcdir}/t.frame
# test description here
#
# test below
# remember to use ${srcdir}
shn=`./spam_header_name`
( echo "$shn: blah1" ; echo ; echo "$shn: blah2" ) >$TMPDIR/input
tail -2 $TMPDIR/input >$TMPDIR/want
../bogofilter -p -e -C <$TMPDIR/input | sed -n '/^$/,$ p' >$TMPDIR/out
cmp $TMPDIR/out $TMPDIR/want
