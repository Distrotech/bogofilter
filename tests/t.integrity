#! /bin/sh

. ${srcdir}/t.frame
# test description here
#
# test below
# remember to use ${srcdir}
printf "test\0te\rst\r\n" >input
../bogofilter -e -p <input >output
head -1 output | cmp - input
