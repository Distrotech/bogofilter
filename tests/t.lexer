#! /bin/sh

. ${srcdir:=.}/t.frame

set -e

unset LANG
unset LC_ALL
unset LC_COLLATE

cat ${srcdir}/inputs/t.lexer.inp | ../bogolexer -p | sort -u > ${TMPDIR}/t.lexer.out

cmp ${srcdir}/outputs/t.lexer.out ${TMPDIR}/t.lexer.out
