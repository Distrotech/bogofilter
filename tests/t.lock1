#! /bin/sh

. ${srcdir:=.}/t.frame

# this test checks that bogofilter -u works
../bogofilter -u -p -e -d $TMPDIR -C </dev/null >/dev/null
