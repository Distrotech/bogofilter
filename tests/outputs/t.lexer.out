account
and
any
anytime
back
behind
biggest
cancelable
cancellation
change
character
city
city's
clancy
clancy's
click
complete
conditions
controversial
credits
crisis
dvds
for
free
giuliani
guided
handling
his
history
inc
including
instructions
jack
junkflix
king's
latest
leadership
learned
length
lessons
mayor
monthly
month-to-month
most
netflix
new
office
partial
period
periods
please
popular
program
rabbit
red
reflects
refunds
rent
reserves
rudolph
ryan
shipping
standard
stephen
story
subject
subscription
terms
the
there
thriller
through
time
tom
trial
visit
want
who
with
www.netflix.com
york
you
your
